File: backend-python/app/routes/accounts.py
# SPDX-License-Identifier: MIT
# Copyright (c) 2025 Gabriel Xia(加百列)
from fastapi import APIRouter, Depends, HTTPException, Request
from ..security import auth_guard
from .auth import users

router = APIRouter()

@router.get('/profile')
async def profile(request: Request, _=Depends(auth_guard)):
    user_id = request.state.user_id
    for u in users.values():
        if u['id'] == user_id:
            return { 'id': u['id'], 'email': u['email'], 'name': u.get('name') }
    raise HTTPException(status_code=401, detail={ 'code': 'UNAUTHORIZED', 'message': 'Not found' })

@router.patch('/profile')
async def update_profile(request: Request, body: dict, _=Depends(auth_guard)):
    name = body.get('name')
    if not name:
        raise HTTPException(status_code=400, detail={ 'code': 'BAD_REQUEST', 'message': 'name required' })
    user_id = request.state.user_id
    for email, u in users.items():
        if u['id'] == user_id:
            users[email]['name'] = name
            return { 'id': u['id'], 'email': u['email'], 'name': name }
    raise HTTPException(status_code=401, detail={ 'code': 'UNAUTHORIZED', 'message': 'Not found' })

File: backend-node/src/routes/accounts.ts
// SPDX-License-Identifier: MIT
// Copyright (c) 2025 Gabriel Xia(加百列)
import { Router, Request, Response } from 'express'
import { authGuard } from '../security/authGuard.js'
import { UsersRepo } from '../data/users.js'

const router = Router()

router.get('/profile', authGuard, (req: Request, res: Response) => {
  const userId = (req as any).userId as string
  const u = UsersRepo.getById(userId)
  if (!u) return res.status(404).json({ ok: false, error: { code: 'NOT_FOUND', message: 'User not found' } })
  res.json({ id: u.id, email: u.email, name: u.name })
})

router.patch('/profile', authGuard, (req: Request, res: Response) => {
  const userId = (req as any).userId as string
  const { name } = req.body || {}
  if (!name || typeof name !== 'string' || name.trim().length === 0) {
    return res.status(400).json({ ok: false, error: { code: 'BAD_REQUEST', message: 'name required' } })
  }
  const u = UsersRepo.updateById(userId, { name: name.trim() })
  if (!u) return res.status(404).json({ ok: false, error: { code: 'NOT_FOUND', message: 'User not found' } })
  res.json({ id: u.id, email: u.email, name: u.name })
})

export const accountsRouter = router
